core/src/main/java/ai/timefold/solver/core/impl/solver/DefaultSolverFactory.java

```diff
@@ -146,7 +146,7 @@ public <Score_ extends Score<Score_>> ScoreDirectorFactory<Solution_, Score_> ge
                 .build();
         var basicPlumbingTermination = new BasicPlumbingTermination<Solution_>(isDaemon);
         var termination = buildTermination(basicPlumbingTermination, configPolicy, configOverride);
-        var phaseList = buildPhaseList(configPolicy, bestSolutionRecaller, termination);
+        var phaseList = buildPhaseList(configPolicy, bestSolutionRecaller, termination, valueRangeManager);
```

Comment:
Why do we need to share the value range manager during the construction of the phases? Can we retrieve it during the lifecycle events?

This appears inconsistent with the approach used by other components of the solver. I feel we are passing information that could be obtained in another way.

Reply:
I was looking for another way. Didn't find any.
The entire value manager handling will need reworking; the current way how it's exposed is too brittle.

Reply:
`MoveStreamsBasedMoveRepository` is invoked during lifecycle actions. Can we retrieve the range manager during these lifecycle actions?

Reply:
The manager is created completely outside of any phase, and then somehow transferred everywhere where it needs to go. This is a horrible thing, and it needs refactoring.

First, we must make sure it is never reused. When we make reuse fail fast, we will immediately see all the places where we need to inject new instances of the manager. And when we know that, we will figure out a proper way to provide those instances.

I can do that in this PR, but it feels like it's already big enough.